openhands/controller/agent_controller.py

```diff
@@ -139,12 +139,12 @@ async def report_error(self, message: str, exception: Exception | None = None):
 
         This method should be called for a particular type of errors, which have:
         - a user-friendly message, which will be shown in the chat box. This should not be a raw exception message.
-        - an ErrorObservation that can be sent to the LLM by the agent, with the exception message, so it can self-correct next time.
+        - an ErrorObservation that can be sent to the LLM by the user role, with the exception message, so it can self-correct next time.
         """
         self.state.last_error = message
         if exception:
             self.state.last_error += f': {exception}'
-        self.event_stream.add_event(ErrorObservation(message), EventSource.AGENT)
+        self.event_stream.add_event(ErrorObservation(message), EventSource.USER)
```

Comment:
In the LLM's context, the pattern is like:
```
role=user, message="your task..."
role=assistant, message="ok! I'm running bash to..."  # action 
role=user, message="the command result or error" # observation
```
observations are "user" messages. It's the response of the environment to the agent action, and it's... not agent. If we could define a role="environment" properly recognized I'd personally embrace it 😅. But I don't think we can with most (or all) hosted llms?

Reply:
Thanks for explanation. Alternatively we could call it environment, and transform it to "user" when sending to LLMs. But anyways that's just unnecessary...

Reply:
Good point, we could use the right source! Actually, I think I'm going to try that... At least it would be clearer in our code.

I could be wrong, I think it _may_ be useful if we can disambiguate for the LLM. One small thing it may help with is [this annoying bit](https://github.com/All-Hands-AI/OpenHands/blob/8a93da51be8059f6ee963900f4289ad5e462bd75/openhands/agenthub/codeact_agent/system_prompt.j2#L46) if the current phrase doesn't address it enough. In addition, IMHO the LLM gets easily confused, and for example when it creates memories from observations, it sometimes says that "the user" did this. Currently a very-draft prompt I'm playing with looks like:
```
When you receive data prefixed with 'Observation from the environment:', it is a non-interactive information from the environment, not from a human user.

"Observation from environment: The system returned a value of 42."
```

Maybe it doesn't matter! So many trivial things matter though.